src/main/java/gregtech/api/recipes/Recipe.java

```diff
@@ -164,8 +164,8 @@ public final boolean matches(boolean consumeIfSuccessful, IItemHandlerModifiable
      * @return true if the recipe matches the given inputs false otherwise.
      */
     public boolean matches(boolean consumeIfSuccessful, List<ItemStack> inputs, List<FluidStack> fluidInputs) {
-        Pair<Boolean, Integer[]> fluids;
-        Pair<Boolean, Integer[]> items;
+        Pair<Boolean, int[]> fluids;
```

Comment:
Might be worth it just to make a class `MatchResult` or something to encapsulate a primitive `boolean + int[]`.

Also, to check if item/fluid inputs are empty first performing routine may be worth it.